src/harness/reference_models/interference/interference.py

```diff
@@ -0,0 +1,669 @@
+
+  Routine to extract CbsdGrantInformation tuple from CBSD data records from 
+  FAD objects
+
+  Args:
+    cbsd_data_records: A list CbsdData object retrieved from FAD records.
+    is_managing_sas: flag indicating cbsd data record is from managing SAS or 
+                     peer SAS
+                     True - Managing SAS, False - Peer SAS
+  Returns:
+    grant_objects: a list of grants, each one being a namedtuple of type 
+                   CBSDGrantInformation, of all CBSDs from SAS UUT FAD and 
+                   SAS Test Harness FAD
+  """
+
+  grant_objects = []
+
+  # Loop over each CBSD grant
+  for cbsd_data_record in cbsd_data_records:
+    registration = cbsd_data_record['registration']
+    grants = cbsd_data_record['grants']
+
+    # Check CBSD location
+    lat_cbsd = registration['installationParam']['latitude']
+    lon_cbsd = registration['installationParam']['longitude']
+    height_cbsd = registration['installationParam']['height']
+    height_type_cbsd = registration['installationParam']['heightType']
+    if height_type_cbsd == 'AMSL':
+      altitude_cbsd = terrainDriver.GetTerrainElevation(lat_cbsd, lon_cbsd)
+      height_cbsd = height_cbsd - altitude_cbsd
+
+    for grant in grants:
+      # Return CBSD information
+      cbsd_grant = CbsdGrantInformation(
+        # Get information from the registration
+        latitude=lat_cbsd,
+        longitude=lon_cbsd,
+        height_agl=height_cbsd,
+        indoor_deployment=registration['installationParam']['indoorDeployment'],
+        antenna_azimuth=registration['installationParam']['antennaAzimuth'],
+        antenna_gain=registration['installationParam']['antennaGain'],
+        antenna_beamwidth=registration['installationParam']['antennaBeamwidth'],
+        cbsd_category=registration['cbsdCategory'],
+        max_eirp=grant['operationParam']['maxEirp'],
+        low_frequency=grant['operationParam']['operationFrequencyRange']['lowFrequency'],
+        high_frequency=grant['operationParam']['operationFrequencyRange']['highFrequency'],
+        is_managed_grant=is_managing_sas)
+      grant_objects.append(cbsd_grant)
+  return grant_objects
+
+
+def computeInterferencePpaGwpzPoint(cbsd_grant, constraint, h_inc_ant, 
+                                  max_eirp, region='SUBURBAN'):
+  """Compute interference grant causes to GWPZ or PPA protection area
+  
+  Routine to compute interference neighborhood grant causes to protection 
+  point within GWPZ or PPA protection area
+
+  Args:
+    cbsd_grant: a namedtuple of type CbsdGrantInformation
+    constraint: protection constraint of type ProtectionConstraint
+    h_inc_ant: reference incumbent antenna height (in meters)
+    max_eirp: The maximum EIRP allocated to the grant during IAP procedure
+    region: Region type of the GWPZ or PPA area
+  Returns:
+    interference: interference contribution(dBm)
+  """
+
+  if (cbsd_grant.latitude == constraint.latitude and
```

Comment:
Info only (nothing to do): I will probably change the prop model so they handle this case, rather than letting this to the client code of the prop model. Will be more robust all over the test harness

Reply:
Agreed. Will keep up with your enhancement. Thanks